File: 30 Days of Code/Python/09 - Day 8 - Dictionaries and Maps.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem
# Difficulty: Easy
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

n = int(input())
d = {}

for i in range(n):
    x = input().split()
    d[x[0]] = x[1]

while True:
    try:
        NAME = input()
        if NAME in d:
            print(NAME, "=", d[NAME], sep="")
        else: print("Not found")
    except:
        break

File: 30 Days of Code/C#/09 - Day 8 - Dictionaries and Maps.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem
// Difficulty: Easy
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

using System;
using System.Collections.Generic;

class Solution {
	static void Main(String[] args) {
		int n = Int32.Parse(Console.ReadLine());
		Dictionary < string,
		string > phonebook = new Dictionary < string,
		string > ();

		for (int i = 0; i < n; i++) {
			string[] line = Console.ReadLine().Split(' ');
			phonebook[line[0]] = line[1];
		}

		string name;
		while ((name = Console.ReadLine()) != null) {
			if (phonebook.ContainsKey(name)) {
				Console.WriteLine(name + "=" + phonebook[name]);
			}
			else {
				Console.WriteLine("Not found");
			}
		}
	}
}
